pandas/core/arrays/string_.py

```diff
@@ -79,37 +84,114 @@ class StringDtype(ExtensionDtype):
     StringDtype
     """
 
-    name = "string"
-
     #: StringDtype.na_value uses pandas.NA
     na_value = libmissing.NA
+    _metadata = ("storage",)
+
+    def __init__(self, storage=None):
+        if storage is None:
+            storage = get_option("mode.string_storage")
```

Comment:
We might want to delay looking up this option and allow the StringDtype to be initialized with `storage=None`. That could eg allow doing `s.astype("string")` preserving the original dtype if the series already is of string dtype (regardless of the default).

Reply:
The latest commit (wip) started to add a parameterised fixture for more testing prior to implementing this.

since the dtype change is user facing this change worthwhile, there is still more to do as many tests use `dtype="string"` but does potentially make this PR harder to review. I will carry on and do the others if doing this makes sense.

Have added `string[pyarrow]` to this fixture. This gives some additional failures for outstanding parts of ArrowStringArray. It maybe that to keep this PR more cleanly scoped, that we defer adding that for a follow-up

Reply:
sure can you open an issue explaining the api you'd like. I have updated _from_seqenece to accept the string "string", but if you are writing code if you do a comparison as a conditional, you expect the storage type to be fixed. and not then do something different. If the only places where this could be usefull is astype, we could put the logic there.